packages/react-devtools-shared/src/devtools/views/Settings/ProfilerSettings.js

```diff
@@ -44,6 +45,28 @@ export default function ProfilerSettings(_: {||}) {
     },
     [store],
   );
+
+  const recordPerfInsightsSubscription = useMemo(
+    () => ({
+      getCurrentValue: () => store.recordPerfInsights,
+      subscribe: (callback: Function) => {
+        store.addListener('recordPerfInsights', callback);
+        return () => store.removeListener('recordPerfInsights', callback);
+      },
+    }),
+    [store],
+  );
+  const recordPerfInsights = useSubscription<boolean>(
+    recordPerfInsightsSubscription,
+  );
```

Comment:
nit: Would be nice to use `useSyncExternalStore` for this instead, e.g.
https://github.com/facebook/react/blob/99a73ac670ce8d71ae8a690e9795d35cbdfa20b6/packages/react-devtools-shared/src/devtools/views/Components/InspectedElement.js#L128-L138

```suggestion
  const recordPerfInsights = useSyncExternalStore(
    function subscribe(callback) {
      store.addListener('recordPerfInsights', callback);
      return function unsubscribe() {
        store.removeListener('recordPerfInsights', callback);
      };
    },
    function getState() {
      return store.recordPerfInsights;
    },
  );
```